src/components/core/Button/Button.tsx

```diff
@@ -1,107 +1,145 @@
-import { memo, useMemo } from 'react';
-import styled from 'styled-components';
-import Loader from '../Loader/Loader';
-import { TitleXS } from '../Text/Text';
+import styled, { css } from 'styled-components';
 import colors from '../colors';
 import transitions from '../transitions';
+import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react';
+import { BODY_FONT_FAMILY } from '../Text/Text';
+import Link from 'next/link';
+import { Spinner } from '../Spinner/Spinner';
 
-type ButtonStyle = 'primary' | 'secondary';
-
-type Props = {
-  type?: ButtonStyle;
-  className?: string;
-  text?: string;
-  mini?: boolean;
-  onClick?: () => void;
-  disabled?: boolean;
-  loading?: boolean;
-  dataTestId?: string;
-};
-
-function Button({
-  type = 'primary',
-  className,
-  text,
-  mini = false,
-  onClick,
-  disabled,
-  loading,
-  dataTestId,
-}: Props) {
-  const ButtonComponent = useMemo(
-    () => (type === 'primary' ? StyledPrimaryButton : StyledSecondaryButton),
-    [type]
-  );
-  return (
-    <ButtonComponent
-      // Renaming this prop `buttonStyle` since the `type` prop
-      // already exists for styled components
-      buttonStyle={type}
-      className={className}
-      onClick={onClick}
-      disabled={disabled}
-      data-testid={dataTestId}
-      mini={mini}
-    >
-      {loading ? (
-        <Loader inverted size={mini ? 'mini' : 'small'} />
-      ) : (
-        <TitleXS color={type === 'primary' ? colors.white : colors.shadow}>{text}</TitleXS>
-      )}
-    </ButtonComponent>
-  );
-}
+// TODO:
+// - why is opacity on disabled+pending so much stronger than just disabled?
```

Comment:
it's really weird but this seems to have something to do with the animation of the spinner inside the button, not sure if this is a browser thing or a styled-components thing

Reply:
seems like a Chrome bug: https://twitter.com/kingersoll/status/1545309693022593024